### PR TITLE
Upgrade httpoison to v1.x

### PR DESCRIPTION
**Pull request by coop**
As discussed in https://github.com/lob/lob-elixir/pull/26 I've renamed the `response` type to `client_response` and changed the `httpoison` dep from `~> 0.x` to `~> 1.x`.

I included the branch in our app and ran our tests - everything looks good. Travis doesn't seem to build (no API key) so not 100% sure there.

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -36,7 +36,7 @@ defmodule Lob.Mixfile do
       {:credo, "~> 0.8", only: [:dev, :test], runtime: false},
       {:dialyxir, "~> 0.5.1", only: [:dev, :test], runtime: false},
       {:excoveralls, "~> 0.7.4", only: :test},
-      {:httpoison, "~> 0.13"},
+      {:httpoison, "~> 1.0"},
```
Comment (coop) [inline mix.exs]: Is there a minimum 1.x version you want to support? I am choosing anything in the 1.x series right now.

**Comment by coop** [comment]
@itsachen I'm closing https://github.com/lob/lob-elixir/pull/26 in favour of this PR.

File: mix.exs (hunk 1 above)
Comment (itsachen) [reply mix.exs]: `1.0` is good for now, looks like `1.2.0` bumps the minimum Elixir version to `1.5` which conflicts with this package's min version.

File: lib/lob/client.ex [hunk 2]
```diff
@@ -11,7 +11,7 @@ defmodule Lob.Client do
 
   @client_version Mix.Project.config[:version]
 
-  @type response :: {:ok, map, list} | {:error, map}
```
Comment (itsachen) [inline lib/lob/client.ex]: Looks like theres some [extra](https://travis-ci.org/lob/lob-elixir/jobs/487207305#L1444) type specs that still reference this:

```sh
$ grep -r "Client.response" .                                                                                                                                                                                     ~/lob/lob-elixir-clones/lob-elixir
./lib/lob/bank_account.ex:  @spec verify(String.t, map, map) :: Client.response
./lib/lob/us_verification.ex:  @spec verify(map, map) :: Client.response
./lib/lob/resource_base.ex:        @spec list(map, map) :: Client.response
./lib/lob/resource_base.ex:        @spec retrieve(String.t, map) :: Client.response
./lib/lob/resource_base.ex:        @spec create(map, map) :: Client.response
./lib/lob/resource_base.ex:        @spec delete(String.t, map) :: Client.response
./lib/lob/us_zip_lookup.ex:  @spec lookup(map, map) :: Client.response
./lib/lob/intl_verification.ex:  @spec verify(map, map) :: Client.response
./lib/lob/us_autocompletion.ex:  @spec autocomplete(map, map) :: Client.response
```

File: mix.exs (hunk 1 above)
Comment (coop) [reply mix.exs]: Great. Is there anything else you would like me to do?

**Comment by coop** [comment]
Ok I’ll sort that out. I haven’t used typespecs before so I was relying on
test failures to show me what was broken.

**Comment by itsachen** [comment]
@coop Merged the branch, will be cutting a version to include it soon

**Pull request merged**